### PR TITLE
Start of docs to inform deploy; python & configs env managemnet

### PR DESCRIPTION
**Pull request by cmharlow**
Included in this PR:

- Python env management additions
   - `.python-version` to indicate the desired and tested version for running this repo
   - `Pipfile` & `Pipfile.lock` for python dependency management (esp. for deploy)
   - `.travis.yml` for Python linting and testing checks. 
- Dev env management additions / updates
   - `.gitignore` for general git management
   - Updated `README.md` for WIP granular install & run steps, starting with local dev environment
- Env configuation proposals
  - created `configs` directory for configurations that may move to Ansible or elsewhere to envs
  - moved `airflow_logrotated` to `configs/lurch` until goes to Airflow or elsewhere specific to envs
  - created `configs/airflow-variables.json` for sample variables used in local dev setup.

### CODE REVIEW

**Comment by cmharlow** [comment]
This is ready for review, albeit the linter fails. It is failing on a known issue we can comment out or repair or whatever.

File: README.md [hunk 1]
```diff
@@ -1,44 +1,244 @@
...
+(cob_datapipeline) $ git clone https://github.com/tulibraries/cob_datapipeline.git
+```
+There are quite a few ways you can get the cob_datapipeline DAGs into that directory; just ensure that the directory mirrors the order seen in the Github repository layout.
+
+**Set DAG Required Airflow Variables for Local Airflow**
+
+To load the required variables into your local airflow, you can perform the following process:
+
+```
+(cob_datapipeline) $ airflow variables -i airflow-variables.json
```
Comment (bibliotechy) [inline README.md]:
```suggestion
(cob_datapipeline) $ airflow variables -i config/airflow-variables.json
```

File: README.md [hunk 2]
```diff
@@ -1,44 +1,244 @@
...
+  [2019-03-21 16:46:26,868] {__init__.py:51} INFO - Using executor SequentialExecutor
+  [2019-03-21 16:46:27,030] {models.py:160} WARNING - cryptography not found - values will not be stored encrypted.
+  13 of 13 variables successfully updated.
+```
+
+You should update that `airflow-variables.json` file to have the variables you need for your local development environment. Note: Airflow does then later update these variables during processes.
```
Comment (bibliotechy) [inline README.md]:
```suggestion
You should update that `airflow-variables.json` file to have the variables you need for your local development environment, particularly the `AIRFLOW_DATA_DIR` `AIRFLOW_HOME` variables, replacing `~/airflow` with the absolute path to your airflow installation.
Note: Airflow does then later update these variables during processes.
```

File: README.md (hunk 1 above)
Comment (cmharlow) [reply README.md]: is in #19

File: README.md (hunk 2 above)
Comment (cmharlow) [reply README.md]: there's nothing particular about those 2, but minus the wording, sure.

**Pull request merged**